src/main/java/com/google/javascript/gents/ModuleConversionPass.java

```diff
@@ -292,192 +407,206 @@ public boolean shouldTraverse(NodeTraversal t, Node n, Node parent) {
+  private void convertRequireToImportsIfImportedIsKeptInJs(ModuleImport moduleImport) {
+    Node nodeToImport = null;
+    // For destructuring imports use `import {foo} from 'goog:bar';`
+    if (moduleImport.isDestructuringImport) {
+      nodeToImport = new Node(Token.OBJECTLIT);
+      for (String localName : moduleImport.localNames) {
+        nodeToImport.addChildToBack(Node.newString(Token.STRING_KEY, localName));
+      }
+      // For non destructuring imports, it is safe to assume there's only one localName
+    } else if (moduleImport.module.hasDefaultExport) {
+      // If it has a default export then use `import foo from 'goog:bar';`
+      nodeToImport = Node.newString(Token.NAME, moduleImport.localNames.get(0));
+    } else {
+      // If it doesn't have a default export then use `import * as foo from 'goog:bar';`
+      nodeToImport = Node.newString(Token.IMPORT_STAR, moduleImport.localNames.get(0));
+    }
+
+    Node importNode =
+        new Node(
+            Token.IMPORT,
+            IR.empty(),
+            nodeToImport,
+            Node.newString("goog:" + moduleImport.requiredNamespace));
+    nodeComments.replaceWithComment(moduleImport.originalImportNode, importNode);
+    compiler.reportChangeToEnclosingScope(importNode);
+
+    for (int i = 0; i < moduleImport.fullLocalNames.size(); i++) {
+      registerLocalSymbol(
+          moduleImport.originalImportNode.getSourceFileName(),
+          moduleImport.fullLocalNames.get(i),
+          moduleImport.requiredNamespace,
+          moduleImport.localNames.get(i));
+    }
+  }
+
+  private void convertRequireForAlreadyConverted(ModuleImport moduleImport) {
+    // we cannot use referencedFile here, because usually it points to the ES5 js file that is
+    // the output of TS, and not the original source TS file.
+    // However, we can reverse map the goog.module name to a file name.
+    // TODO(rado): sync this better with the mapping done in tsickle.
+    String originalPath =
+        moduleImport.requiredNamespace.replace(alreadyConvertedPrefix + ".", "").replace(".", "/");
+    String referencedFile =
+        pathUtil.getImportPath(moduleImport.originalImportNode.getSourceFileName(), originalPath);
     // case of side-effectful imports.
     // goog.require('...'); -> import '...';
     Node importSpec = IR.empty();
-    if (n.getFirstChild() != null && n.getFirstChild().isDestructuringLhs()) {
+    Node requireLHS = moduleImport.originalImportNode.getFirstChild();
+    if (moduleImport.isDestructuringImport) {
       importSpec = new Node(Token.IMPORT_SPECS);
-      for (String fullLocalName : fullLocalNames) {
+      for (String fullLocalName : moduleImport.fullLocalNames) {
         importSpec.addChildToBack(IR.name(fullLocalName));
       }
-    } else if (n.getFirstChild() != null && n.getFirstChild().isName()) {
+    } else if (requireLHS != null && requireLHS.isName()) {
```

Comment:
moduleImport can have a method for this too - maybe 'isFullModuleImport'?

Reply:
Done.